packages/forklift-console-plugin/src/modules/Providers/views/details/components/DetailsSection/components/UIDetailsItem.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useModal } from 'src/modules/Providers/modals';
+import { EditProviderUIModal } from 'src/modules/Providers/modals/EditProviderUI';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { ExternalLink } from '@kubev2v/common';
+import { V1beta1Provider } from '@kubev2v/types';
+
+import { DetailsItem } from '../../../../../utils';
+
+import { ProviderDetailsItemProps } from './ProviderDetailsItem';
+
+/**
+ * @typedef {Object} UIDetailsItemProps - extends ProviderDetailsItemProps
+
+ * @property {string} [text] - A text to be displayed as a content.
+ * @property {function} [calcProviderUI] - A method for auto calculating the provider's UI path.
+ **/
+export interface UIDetailsItemProps extends ProviderDetailsItemProps {
+  text: string;
+  calcProviderUI: (provider: V1beta1Provider) => string;
+}
+
+const getProviderUIAnnotation = (provider: V1beta1Provider): string =>
+  provider?.metadata?.annotations?.['forklift.konveyor.io/providerUI'];
+
+export const UIDetailsItem: React.FC<UIDetailsItemProps> = ({
+  resource: provider,
+  canPatch,
+  text,
+  calcProviderUI,
+}) => {
+  const { t } = useForkliftTranslation();
+  const { showModal } = useModal();
+  // Read the annotation and only if empty, try to auto calculate the provider UI path
+  const content =
+    getProviderUIAnnotation(provider) ?? (calcProviderUI ? calcProviderUI(provider) : null);
+
+  return (
+    <DetailsItem
+      content={
+        content ? (
+          <ExternalLink text={text} href={content} isInline={true}>
+            {content}
+          </ExternalLink>
+        ) : (
+          <span className="text-muted">
+            {t('Optional: Click the edit button for setting the provider web UI path')}
+          </span>
+        )
+      }
+      onEdit={
+        canPatch &&
+        provider?.metadata &&
+        (() => showModal(<EditProviderUIModal resource={provider} content={content} />))
```

Comment:
```suggestion
        (() => undefined)
```
this is not implemented yet ...

Reply:
and then you don't need to add a null modal ...

Reply:
I don't want to add it for the 1st  provider's implementation  if I already prepare this infra PR then why not preparing it here so that for each provider, I'll only touch the `EditProviderUIModal`  implementation. And won't touch this component.
But it's not important. I can also add it for the 1st provider.